src/interface/src/app/treatments/treatment-project-area/treatment-project-area.component.ts

```diff
@@ -51,4 +67,86 @@ export class TreatmentProjectAreaComponent implements OnDestroy {
     this.selectedStandsState.clearStands();
     this.treatmentsState.setShowApplyTreatmentsDialog(false);
   }
+
+  treatmentPlanId: number = this.route.snapshot.data['treatmentId'];
+  projectAreaId: number = this.route.snapshot.data['projectAreaId'];
+
+  treatmentPlan: TreatmentPlan | null = null;
+  notesModel = 'project_area';
+
+  ngOnInit(): void {
+    if (this.treatmentPlanId) {
+      this.treatmentsService
+        .getTreatmentPlan(Number(this.treatmentPlanId))
+        .subscribe((r) => (this.treatmentPlan = r));
+    }
+    this.loadNotes();
+  }
+
+  // notes data
+  notes: Note[] = [];
+  notesSidebarState: NotesSidebarState = 'READY';
+
+  //notes handling functions
+  addNote(comment: string) {
+    this.notesSidebarState = 'SAVING';
+    if (this.projectAreaId) {
+      this.notesService.addNote(this.projectAreaId, comment).subscribe({
+        next: () => {
+          this.loadNotes();
+        },
+        error: () => {
+          this.snackbar.open(
+            `Error: Could not add note.`,
+            'Dismiss',
+            SNACK_ERROR_CONFIG
+          );
+        },
+        complete: () => {
+          this.notesSidebarState = 'READY';
+        },
+      });
+    }
+  }
+
+  handleNoteDelete(note: Note) {
+    const dialogRef = this.dialog.open(DeleteNoteDialogComponent, {});
+    dialogRef
+      .afterClosed()
+      .pipe(take(1))
+      .subscribe((confirmed) => {
+        if (confirmed) {
+          this.notesService.deleteNote(this.projectAreaId, note.id).subscribe({
+            next: () => {
+              this.snackbar.open(
+                `Deleted note`,
+                'Dismiss',
+                SNACK_NOTICE_CONFIG
+              );
+            },
+            error: () => {
+              this.snackbar.open(
+                `Error: Could not delete note.`,
+                'Dismiss',
+                SNACK_ERROR_CONFIG
+              );
+            },
+            complete: () => {
+              this.loadNotes();
+            },
+          });
+        }
+      });
+  }
+
+  loadNotes() {
+    if (this.projectAreaId) {
+      this.notesService
+        .getNotes(this.projectAreaId)
+        .subscribe((notes: Note[]) => {
+          this.notes = notes;
+          this.cdr.detectChanges();
```

Comment:
also forgot to mention, but why do you need this.cdr.detectChanges();? this is usually a tell-tale of something else being wrong, unless you are using dependencies that run outside angular cycle, which this one doesnt seem to be the case?

Reply:
This was intermittently slow. I don't think changes to `this.notes` was forcing a redraw, but using spread seemed to resolve it.